### PR TITLE
Add Apache 2.0 and MIT License to XMSS

### PR DESCRIPTION
**Pull request by ducnguyen-sb**
XMSS software will be licensed under dual license: Apache 2.0 and MIT.

### CODE REVIEW

**Comment by ducnguyen-sb** [comment]
If no one has any problem with the dual license, I'll merge this PR. 

I think it's best to set under dual license, to maximize the adoption opportunity.

If anyone has a reason to set from `Apache 2.0 AND MIT` to `Apache 2.0 OR MIT`, let me know.

File: src/sig_stfl/xmss/LICENSE-MIT [hunk 1]
```diff
@@ -0,0 +1,9 @@
+The MIT License (MIT)
+
+Copyright © 2024 SandboxAQ
```
Comment (baentsch) [inline src/sig_stfl/xmss/LICENSE-MIT:3]: I acknowledge all the work done by SandboxAQ and the company's copyright ownership is implied, but I think this is the first time any company puts its "ownership stamp" so blatantly onto code within OQS that I'd like to ask the whole team whether we want to move forward with this approach? What should motivate independent contributors not paid for any of this keep working on (and if it's just reviewing) code bases labelled some company's "property"?

Comment (dstebila) [reply src/sig_stfl/xmss/LICENSE-MIT:3]: The first line of the [MIT license](https://en.wikipedia.org/wiki/MIT_License) is `Copyright (c) <year> <copyright holders>` so I think this is in agreement with that. We don't require contributors to assign copyright to the project; they hold their own copyright, and license it to the project under the MIT License.

Comment (ducnguyen-sb) [reply src/sig_stfl/xmss/LICENSE-MIT:3]: I could just give it a link to MIT license as in the website: https://mit-license.org/, 
but then I remember when creating a MIT license in Github, it requires to put a copyright holder, for no particular reason I put in the company name to hope the it's a valid MIT license. 

I also look around at liboqs code base for MIT license, for example: https://github.com/open-quantum-safe/liboqs/blob/c2c969c02889f2766d78a74ffb7d4c3a6dab9cd5/src/kem/kyber/oldpqclean_kyber768_aarch64/macros_common.inc#L7C74-L7C74. 

For the license part, `Apache 2.0 AND MIT` just for maximize adoption. If liboqs want it to be `Apache 2.0 OR MIT`, just let me know. Or Just `Apache 2.0` or `MIT`, just let me know. 

And for MIT license, I can remove the copyright part, but I wonder if I remove the copyright, would it be a valid MIT license? I genuinely don't know.

Comment (baentsch) [reply src/sig_stfl/xmss/LICENSE-MIT:3]: > I wonder if I remove the copyright, would it be a valid MIT license? I genuinely don't know.

Not being a lawyer I don't know either for sure but guess it should be there. If reference were given to the OQS project or individuals doing the work (as in other instances and as [I personally have always done it](https://github.com/open-quantum-safe/oqs-provider/blob/eb80076d1683121db35fabf8c1b0c6f9125c1b6d/LICENSE.txt#L1-L3)) I would not have commented.

Now, given that 
- you as an individual are not holder of the copyright but your employer is, and 
- that the OQS project does not require Copyright transfer by the contributor

I acknowledge that your license is correct and my comment was wrong. Apologies. 

I am probably a bit touchy these days that for-profit companies by way of LinuxFoundation take control of a project I felt had been operating for the common good. The latter had been my prime motivation for working on it. Continuing this way --and for free-- just makes me feel stupid.

Comment (ducnguyen-sb) [reply src/sig_stfl/xmss/LICENSE-MIT:3]: @baentsch  No problem. Thank you for reviewing my PR.

**Comment by ducnguyen-sb** [comment]
@dstebila @SWilson4 : Can you review this PR ?

**Review by SWilson4 — APPROVED** [review]

**Pull request merged**